src/actions/proxy.ts

```diff
@@ -0,0 +1,129 @@
+import { Constructor, Tinlake } from '../types';
+import { waitAndReturnEvents, executeAndRetry } from '../ethereum';
+const abiCoder = require('web3-eth-abi');
+import BN from 'bn.js';
+
+function ProxyActions<ActionsBase extends Constructor<Tinlake>>(Base: ActionsBase) {
+
+  return class extends Base implements IProxyActions {
+
+    shelf = this.contracts['SHELF'].address;
+    currency = this.contracts['TINLAKE_CURRENCY'].address;
+    registry = this.contracts['COLLATERAL_NFT'].address;
+    actions = this.contracts['ACTIONS'].address;
+    proxy = this.eth.contract(this.contractAbis['PROXY']).at(this.ethConfig.proxy)
```

Comment:
the proxy,  proxy registry, actions, contract should be added in Tinlake.ts along with the other contracts and be part of contracts class var